### PR TITLE
chore(bottlecap): reduce lock contention in logs

### PR DESCRIPTION
**Pull request by duncanista**
# What?

1. Drop logs `Aggregator` lock as soon as possible.
2. Make the Telemetry API listener to send the events directly to the logs agent, and send events of interest to the main thread.
- Previously, we were sending first the events to the main thread event bus, which was bounded to 100 elements, and we had to wait for them to be processed there to be forwarded to the logs agent. This is not ideal.

### CODE REVIEW

File: bottlecap/src/logs/aggregator.rs [hunk 1]
```diff
@@ -1,11 +1,10 @@
-use serde::Serialize;
-use std::collections::VecDeque;
+use crossbeam::queue::SegQueue;
```
Comment (blt) [inline bottlecap/src/logs/aggregator.rs:1]: FWIW SegQueue is an unbounded multi-producer, multi-consumer. Minor issue, the multi-consumer bit requires more coordination in the implementation so you're leaving performance on the table by not using a single-consumer. More pressing, this is an unbounded queue and represents a potential site for unbounded allocations.

Comment (duncanista) [reply bottlecap/src/logs/aggregator.rs:1]: I'm aware of the tradeoffs, I have to clean this PR still, will mark it as draft

Comment (blt) [reply bottlecap/src/logs/aggregator.rs:1]: SegQueue is also theoretically lock free but it uses Acquire/Release semantics internally and so the semantics are equivalent to a mutex. Except Acq/Rel done on atomic structures cuts the Linux scheduler out so your latency will be worse generally speaking, except in specialized circumstances where the acq/rel using software has the highest CPU priority anyway and is always scheduled first. 

Tokio's mpsc is good, will play well with the Linux scheduler and also nicely between async/sync code.

**Comment by duncanista** [comment]
Closing due to other PRs doing this

**Pull request closed without merging**